addons/im_livechat/models/discuss_channel.py

```diff
@@ -252,11 +243,12 @@ def _message_post_after_hook(self, message, msg_vals):
         return super()._message_post_after_hook(message, msg_vals)
 
     def _chatbot_restart(self, chatbot_script):
-        self.write({
+        # sudo: discuss.channel - visitor can clear current step to restart the script
+        self.sudo().write({
             'chatbot_current_step_id': False
         })
```

Comment:
could take the opportunity to make this a single line I guess (and maybe use field = syntax ?)